crates/iroha_core/src/sumeragi/main_loop.rs

```diff
@@ -262,11 +262,11 @@ impl Sumeragi {
                         }
                     };
 
-                    if block.as_ref().errors().next().is_some() {
+                    if let Some(error) = block.as_ref().errors().next() {
                         error!(
                             peer_id=%self.peer,
                             role=%self.role(),
-                            "Genesis contains invalid transactions"
+                            ?error
                         );
```

Comment:
Actually, would be useful to print ALL errors collected. And "genesis contains invalid transaction" is a useful addition too.